Jellyfin/Views/VideoViewModel.cs

```diff
@@ -70,12 +81,134 @@ public async void HandleParameters(Video.Parameters parameters)
             _playerElement.SetMediaPlayer(new MediaPlayer());
             _playerElement.MediaPlayer.Source = MediaSource.CreateFromAdaptiveMediaSource(ams);
             _playerElement.MediaPlayer.Play();
+            
 
             ams.InitialBitrate = ams.AvailableBitrates.Max();
+
+            _playerElement.MediaPlayer.MediaEnded += (mp, o) =>
+            {
+                _ = _jellyfinApiClient.UserPlayedItems[videoId].PostAsync();
+            };
+            _playerElement.MediaPlayer.PlaybackSession.PlaybackStateChanged += async (session, obj) =>
+            {
+                switch (session.PlaybackState)
+                {
+                    case MediaPlaybackState.Paused:
+                        _ = _jellyfinApiClient.Sessions.Playing.Progress.PostAsync(new PlaybackProgressInfo()
+                        {
+                            ItemId = videoId,
+                            MediaSourceId = videoId.ToString("N"),
+                            AudioStreamIndex = parameters.AudioStreamIndex,
+                            SubtitleStreamIndex = parameters.SubtitleStreamIndex,
+                            PlaySessionId = _playingSessionId,
+                            PositionTicks = await GetCurrentTicks(),
+                            SessionId = _appSettings.SessionId,
+                            IsPaused = true
+                        });
+                        break;
+                    case MediaPlaybackState.Playing:
+                        _ = _jellyfinApiClient.Sessions.Playing.Progress.PostAsync(new PlaybackProgressInfo()
+                        {
+                            CanSeek = true,
+                            ItemId = videoId,
+                            MediaSourceId = videoId.ToString("N"),
+                            AudioStreamIndex = parameters.AudioStreamIndex,
+                            SubtitleStreamIndex = parameters.SubtitleStreamIndex,
+                            PlaySessionId = _playingSessionId,
+                            PositionTicks = await GetCurrentTicks(),
+                            SessionId = _appSettings.SessionId,
+                            IsPaused = false
+                        });
+                        break;
+                }
+            };
+
+            await _jellyfinApiClient.PlayingItems[videoId].PostAsync(request =>
+            {
+                request.QueryParameters.MediaSourceId = videoId.ToString("N");
+                request.QueryParameters.AudioStreamIndex = parameters.AudioStreamIndex;
+                request.QueryParameters.SubtitleStreamIndex = parameters.SubtitleStreamIndex;
+                request.QueryParameters.PlaySessionId = _playingSessionId = playbackInfo.PlaySessionId;
```

Comment:
Seems like you can set `_playingSessionId` as soon as we have the `playbackInfo` instead of delaying it until here (and setting it again in the Timer below?